docs/docs/vm-fundamentals/hints.md

```diff
@@ -2,6 +2,137 @@
 sidebar_position: 3
 ---
 
-# Hints
 
-There are several hints, explain each of them here. How they interact with the VM and affects it.
+# Hint
+  ## HINTS  IN CAIRO VIRTUAL MACHINE
+Hints in Cairo are pieces of Python code only executed by the sequencer.
+
+Hints are used to help the Virtual Machine determine the next path of execution it should take.
+
+Below are some Cairo code with hints:
+
+```cairo
+      [ap] = 25, ap++;
+    %{
+        import math
+         memory[ap] = int(math.sqrt(memory[ap - 1]))
+     %}
+      [ap - 1] = [ap] * [ap], ap++;
+```
+
+
+```cairo
+      func main() {
+          %{ memory[ap] = program_input['secret'] %}
+         [ap] = [ap], ap++;
+         ret;
+      }
+``` 
+
+
+
+### WHY HINTS RUN ON SEQUENCERS/ PROVERS AND NOT VERIFIERS
+
+Sequencers are the leads that produces blocks by executing transaction and updating the blockchain state.
+During the proof generation process, the prover uses hints to generate additional constraints or assumptions about the program's behavior, which makes part of the proof sent to the verifier for verification. In the Cairo Virtual Machine, hints are sorted out during different stages of the compilation and execution process by the sequencer. However, they are not directly involved in the verification stage. Hints are primarily for performance optimization.
```

Comment:
During proof generating, hints are not used at all. Hints are only used when generating the execution trace. This execution trace is then proven by the Prover but at this stage it doesn't care about hints. Again, please don't use information you're not certain of. 

Let's remove this sentence.